sklearn/manifold/spectral_embedding_.py

```diff
@@ -47,17 +47,18 @@ def _graph_connected_component(graph, node_id):
     nodes_to_explore = np.zeros(shape=(graph.shape[0]), dtype=np.bool)
     nodes_to_explore[node_id] = True
     n_node = graph.shape[0]
-    nodes_to_add = np.zeros(shape=(graph.shape[0]), dtype=np.bool)
     for i in range(n_node):
```

Comment:
To help readability we could write `for _ in range(n_node):` since the index is not used